### DISCUSSION TITLE
v9: IsBackOfficeRequest is extremely inefficient

### DISCUSSION THREAD

**Discussion by Nuklon** [umbraco/Umbraco-CMS#11640] [Features and ideas]
https://github.com/umbraco/Umbraco-CMS/blob/5bfab13dc5a268714aad2426a2b68ab5561a6407/src/Umbraco.Core/Routing/UmbracoRequestPaths.cs#L72
This gets called a lot and does a lot of unnecessary things.
1) It can optionally take the Path property from a Request instead of trimming the entire URL.
2) _defaultUmbPaths can be a hashset.
3) urlPaths get split by forward slash for PluginController, while this should just count slashes and avoid unnecessary allocations.

**Comment by Nuklon** [comment]
@bergmania Not sure how to call for this but this should've been created as an issue instead.

**Comment by p-m-j** [comment]
IIRC this is mostly used to work out whether or not to read the backoffice auth ticket, which we should probably just always do when present.

Method is internal so we can nuke it pretty easily.

**Comment by Zeegaan** [comment]
Locking this discussion, follow the progres on the issue here: https://github.com/umbraco/Umbraco-CMS/issues/13732
